### PR TITLE
refactor: migrate gradle from groovy to kotlin

### PR DESCRIPTION
**Pull request by iProdigy**
* Groovy -> Kotlin
* Bump lombok
* Bump shadow
* Bump guava
* Use JUnit bom
* Bump gradle

### CODE REVIEW

**Review by pajlada — APPROVED** [review]
Looks good to me, I'll let @Mm2PL merge this in once he's had a chance to look at it

Everything runs and works as expected

**Review by Mm2PL — APPROVED** [review]
seems good

**Pull request merged**
